### ISSUE TITLE
GridModel should still allow custom pinned rows

### ISSUE THREAD

**Issue by ghsolomon** [xh/hoist-react#3259]
Developers should be able to utilize AG-Grid's support for custom pinned rows, but currently `GridModel` overwrites pinned row data provided via `agOptions`.  As a workaround, pinned rows can be set by calling `agApi.setPinnedTopRowData()`, but currently the data provided must be `StoreRecords`, which is not ideal because we don't expect application code to construct `StoreRecords`.  Consider changing Hoist methods that deal with pinned row data to first confirm that the data is `instanceof StoreRecord`.

**Comment by haynesjm42** [comment]
Not exactly the same issue @ghsolomon ran into but I ran into some pain points today due to our lack of proper pinned row support, specifically around getting our autosize process to recognize the values in the pinned rows, which it wont out of the box since they are not records in the store. Ended up getting something working with some hacky code using the ag-grid external filter, but it's really not ideal and probably will introduce other issues down the line since these pinned rows are now technically in my store but not visible in the grid.

Was thinking about how we might support this, and I think the best approach would be for GridModel to have 2 additional stores - one for the top pinned records and one for the bottom records. We could create these automatically using the main store as a template, and provide methods on GridModel to load/update/etc records into the appropriate store, which we would then pull from for top and bottom pinned row data. GridAutosizeService could then include the records from these stores in it's width calculations.

If we do this then I think we can just drop any kind of support for pinned row data provided directly through the ag-grid api, which would be a breaking change for any apps which use it, though I think that is a worthwhile breaking change to bring pinned row data in line with normal grid row data
